core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/RDDPlugin.scala

```diff
@@ -39,14 +39,22 @@ class RDDPlugin(
 
   override def rddReadNodeProcessor: PartialFunction[RDD[_], ReadNodeInfo] = {
     case fsr: FileScanRDD =>
-      val uris = fsr.filePartitions.flatMap(_.files.map(_.filePath))
+      val files = fsr.filePartitions.flatMap(_.files)
+      val uris = files.map(extractPath(_))
       ReadNodeInfo(SourceIdentifier(None, uris: _*), Map.empty)
     case hr: HadoopRDD[_, _] =>
       val partitions = ReflectionUtils.extractValue[Array[Partition]](hr, "partitions_")
       val uris = partitions.map(hadoopPartitionToUriString)
       ReadNodeInfo(SourceIdentifier(None, uris: _*), Map.empty)
   }
 
+  private def extractPath(file: PartitionedFile): String = {
```

Comment:
These were some changes from the original PR #739 that this was based off of. I read mention of potential binary incompatibility yet these tests passed under both 3.3 + 3.4 profile. Any thoughts here?

Reply:
I think the changes from #739 are fine if the tests pass, no need to change it.

Reply:
Sounds good!